app/assets/javascripts/application.js

```diff
@@ -3,6 +3,94 @@
 //= require turbolinks
 //= require_tree .
 
+// videocall
+
+let remoteContainer;
+let canvasContainer;
+
+let handleFail = function (err) {
+	console.log("Error : ", err);
+};
+
+function addVideoStream(streamId) {
+	let streamDiv = document.createElement("div");
+	streamDiv.id = streamId;
+	streamDiv.style.transform = "rotateY(180deg)";
+	remoteContainer.appendChild(streamDiv);
+}
+
+function removeVideoStream(evt) {
+	let stream = evt.stream;
+	stream.stop();
+	let remDiv = document.getElementById(stream.getId());
+	remDiv.parentNode.removeChild(remDiv);
+	console.log("Remote stream is removed " + stream.getId());
+}
+
+function addCanvas(streamId) {
+	let canvas = document.createElement("canvas");
+	canvas.id = 'canvas' + streamId;
+	canvasContainer.appendChild(canvas);
+	let ctx = canvas.getContext('2d');
+	let video = document.getElementById(`video${streamId}`);
+
+	video.addEventListener('loadedmetadata', function () {
+		canvas.width = video.videoWidth;
+		canvas.height = video.videoHeight;
+	});
+
+	video.addEventListener('play', function () {
+		var $this = this;
+		(function loop() {
+			if (!$this.paused && !$this.ended) {
+				ctx.drawImage($this, 0, 0);
+				setTimeout(loop, 1000 / 30);
+			}
+		})();
+	}, 0);
+}
+
+let client = AgoraRTC.createClient({
+	mode: 'live',
+	codec: "h264"
+});
+
+client.init("5b4594eea0de44a39e6a3ce3605d27b8", () => console.log("AgoraRTC client initialized"), handleFail);
+
+client.join(null, "any-channel", null, (uid) => {
+
+	let localStream = AgoraRTC.createStream({
+		streamID: uid,
+		audio: false,
```

Comment:
What does this do? Doesn't that mean that we have no audio?

Reply:
We are not 100% sure before we can see it on Heroku, but it means we have no audio track for the videocall. We have changed it true for now.